### PR TITLE
Add searching for posts functionality.

### PR DESCRIPTION
**Pull request by bobbyshaw**

### CODE REVIEW

File: code/Model/Post.php [hunk 1]
```diff
@@ -192,6 +192,58 @@ public function fetchByUserId($userId)
...
+    /**
+     * Search string within post title and body and author name.
+     * @param $term
+     *
+     * @return array
+     */
+    public function fetchByTerm($term)
+    {
+
+        /**
```
Comment (kalenjordan) [inline code/Model/Post.php]: @bobbyshaw maybe we could put this comment into the docblock?

File: code/Controller/Search.php [hunk 2]
```diff
@@ -4,7 +4,13 @@ class Controller_Search extends Controller_Abstract
 {
     public function get()
     {
+
```
Comment (kalenjordan) [inline code/Controller/Search.php]: @bobbyshaw kind of a pet peeve but I'm not really into the leading newline character at the top of the method body :)  I know some people like that.  Maybe we can pick a code style / standard / sniffer or something and just go with that.

**Comment by kalenjordan** [comment]
Took a really quick look at the code - I don't worry too much about your PRs though :)  Go ahead and merge.  It won't let me merge the PR right now - might need to rebase or something...

File: code/Controller/Search.php (hunk 2 above)
Comment (bobbyshaw) [inline code/Controller/Search.php]: Sure thing. It wasn't done intentionally.

File: code/Model/Post.php (hunk 1 above)
Comment (bobbyshaw) [inline code/Model/Post.php]: Just the algorithm description or intended sql output as well?

Comment (kalenjordan) [inline code/Model/Post.php]: I think this is just a personal preference of mine - I usually like to put any kind of /** comment inside of a method docblock.  Not a big deal either way.  If there's some kind of specific code sniffer we can subscribe to, that might simplify things.

**Comment by bobbyshaw** [comment]
Available on staging to test.

**Pull request merged**